src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -2761,6 +2761,39 @@ class X
             );
     }
 
+    [Fact, WorkItem(59466, "https://github.com/dotnet/roslyn/issues/59466")]
+    public void AlwaysTruePattern()
+    {
+        var source = @"
+_ = new S() is [..var y];
+_ = new S() is [..];
+_ = new S() is [..[..]];
+_ = new S() is not [..];
+
+struct S
+{
+    public int Length => 1;
+    public int this[int i] => 42;
+    public S this[System.Range r] => default;
+}
+";
+        var comp = CreateCompilationWithIndexAndRangeAndSpan(source);
+        comp.VerifyEmitDiagnostics(
+            // (2,5): warning CS8794: An expression of type 'S' always matches the provided pattern.
+            // _ = new S() is [..var y];
+            Diagnostic(ErrorCode.WRN_IsPatternAlways, "new S() is [..var y]").WithArguments("S").WithLocation(2, 5),
```

Comment:
Is the expectation here that user should simplify to `var y`?

Reply:
The user should simply write `is var y` (which would not warn).
List and slice patterns make assumptions about collections being well-behaved. If you grab the entire range, you should get the same collection, so that dicing wasn't necessary.